src/check-dsl.ts

```diff
@@ -52,6 +55,325 @@ const defaultError = (lines: string[]) => {
   };
 };
 
+// helper function to figure out whether the specified allowable types
+// are tuple to user set.  If so, return the type and relationship.
+// Otherwise, return null as relationship
+function destructTupleToUserset(allowableType: string): string[] {
+  return allowableType.split("#", 2);
+}
+
+// helper function to parse thru a child relation to see if there are unique entry points
```

Comment:
can you be more descriptive here? what is a unique entrypoint? maybe give an example?